lib/improver/spotdata/extrema.py

```diff
@@ -0,0 +1,249 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Calculate extrema values for diagnostics."""
+
+import numpy as np
+import numpy.ma as ma
+import datetime
+from datetime import datetime as dt
+import iris
+from iris.cube import Cube, CubeList
+from iris.analysis import MAX as IMAX
+from iris.analysis import MIN as IMIN
+from iris.coords import DimCoord
+from improver.spotdata.common_functions import (iris_time_to_datetime,
+                                                datetime_constraint,
+                                                dt_to_utc_hours)
+
+
+class ExtractExtrema(object):
+    """Extract diagnostic maxima and minima in a given time period."""
+
+    def __init__(self, period, start_hour=9):
+        """
+        The class is used to calculate maxima and minima values of a diagnostic
+        over the supplied period (in hours), starting from a given hour in the
+        24hr clock. All extrema values are calculated in the local time of the
+        site being considered.
+
+        Args:
+        -----
+        period : int
+            Period in hours over which to calculate the extrema values, e.g.
+            24 hours for maxima/minima in a whole day.
+
+        start_hour : int
+            Hour in the 24hr clock at which to start the series of periods,
+            e.g. period=12, start_hour=9 --> 09-21, 21-09, etc.
+
+        """
+        self.period = period
+        self.start_hour = start_hour
+
+    def process(self, cube):
+        """
+        Calculate extrema values for diagnostic in cube over 24 hour period.
```

Comment:
I think you mean that the extrema values are calculated over the period specified in the initialisation of the class, rather than for a 24 hour period.